test/OpenTelemetry.Tests/Implementation/Trace/Propagation/CompositePropagatorTest.cs

```diff
@@ -0,0 +1,193 @@
+            var carrier = new Dictionary<string, string>();
+            compositePropagator.Inject(activityContext, carrier, Setter);
+
+            Assert.Equal(expectedHeaders, carrier);
+
+            var ctx = compositePropagator.Extract(activityContext, expectedHeaders, Getter);
+            Assert.Equal(activityContext.TraceId, ctx.TraceId);
+            Assert.Equal(activityContext.SpanId, ctx.SpanId);
+            Assert.True(ctx.IsValid());
+
+            bool isInjected = compositePropagator.IsInjected(carrier, Getter);
+            Assert.True(isInjected);
+        }
+
+        [Fact]
+        public void CompositePropagator_B3FormatNotInjected()
+        {
+            var carrier = new Dictionary<string, string>
+            {
+                { TraceParent, $"00-{this.traceId}-{this.spanId}-01" },
+            };
+
+            var compositePropagator = new CompositePropagator(new List<ITextFormat>
+            {
+                new TraceContextFormat(),
+                new B3Format(),
+            });
+
+            bool isInjected = compositePropagator.IsInjected(carrier, Getter);
+            Assert.False(isInjected);
+        }
+
+        [Fact]
+        public void CompositePropagator_TestPropagator()
+        {
+            var compositePropagator = new CompositePropagator(new List<ITextFormat>
+            {
+                new TestPropagator("custom-traceparent-1", "custom-tracestate-1"),
+                new TestPropagator("custom-traceparent-2", "custom-tracestate-2"),
+            });
+
+            var activityContext = new ActivityContext(this.traceId, this.spanId, ActivityTraceFlags.Recorded, traceState: null);
+            var carrier = new Dictionary<string, string>();
+
+            compositePropagator.Inject(activityContext, carrier, Setter);
+            Assert.Contains(carrier, kv => kv.Key == "custom-traceparent-1");
+            Assert.Contains(carrier, kv => kv.Key == "custom-traceparent-2");
+
+            bool isInjected = compositePropagator.IsInjected(carrier, Getter);
+            Assert.True(isInjected);
+        }
+
+        [Fact]
+        public void CompositePropagator_UsingSameTag()
+        {
+            var compositePropagator = new CompositePropagator(new List<ITextFormat>
+            {
+                new TestPropagator("custom-traceparent", "custom-tracestate-1"),
+                new TestPropagator("custom-traceparent", "custom-tracestate-2"),
+            });
+
+            var activityContext = new ActivityContext(this.traceId, this.spanId, ActivityTraceFlags.Recorded, traceState: null);
+            var carrier = new Dictionary<string, string>();
+
+            compositePropagator.Inject(activityContext, carrier, Setter);
+            Assert.Contains(carrier, kv => kv.Key == "custom-traceparent");
+
+            bool isInjected = compositePropagator.IsInjected(carrier, Getter);
+            Assert.True(isInjected);
```

Comment:
Lets validate that while injecting with 2 propagators using same key, the last one wins? There is no requirement that the 2nd one checks that header already exist and backs off right?

Same with extract - validate that we return when we find the 1st valid context.

Reply:
For the inject part: ok, i did some changes that will validate the value and check if the latest one is the one that we have.
For the isInjected part: all will have to be injected. Since i did one with same header and one with different ones. i think we are covered.
For the extract part: changed the constructor to pass a boolean, so we can define if we will return a default or not. With that in mind, I counted how many times we used the Getter. So, if the first one returns default, we have to access only two times.